test/hotspot/jtreg/testlibrary_tests/template_framework/examples/TestTutorial.java

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8344942
+ * @summary Demonstrate the use of Templates with the Compile Framework.
+ *          It displays the use of most features in the Template Framework.
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib /
+ * @run main template_framework.examples.TestTutorial
+ */
+
+package template_framework.examples;
+
+import java.util.Collections;
+import java.util.List;
+
+import compiler.lib.compile_framework.*;
+
+import compiler.lib.template_framework.Template;
+import compiler.lib.template_framework.Hook;
+import compiler.lib.template_framework.TemplateBinding;
+import compiler.lib.template_framework.Name;
+import static compiler.lib.template_framework.Template.body;
+import static compiler.lib.template_framework.Template.let;
+import static compiler.lib.template_framework.Template.$;
+import static compiler.lib.template_framework.Template.fuel;
+import static compiler.lib.template_framework.Template.addName;
+import static compiler.lib.template_framework.Template.sampleName;
+import static compiler.lib.template_framework.Template.weighNames;
+
+import compiler.lib.template_framework.library.Hooks;
+
+public class TestTutorial {
+
+    public static void main(String[] args) {
+        // Create a new CompileFramework instance.
+        CompileFramework comp = new CompileFramework();
+
+        // Add java source files.
```

Comment:
Maybe it would also be nice to see the actually generated strings for the templates. Should we add an easy way to do this just for the tutorials in this file? Maybe we can do it by asking the user to pass an environment property like `-DPrintTemplates=true` or something like that. Or is there already a way provided by the framework to print the resulting templates on demand?

Reply:
There is `-DCompileFrameworkVerbose=true`, which will print the code that the CompileFramework compiles. I think that would be good enough, right?